consensus/module.go

```diff
@@ -149,6 +150,10 @@ func (*consensusModule) Create(bus modules.Bus, options ...modules.ModuleOption)
 
 	consensusCfg := runtimeMgr.GetConfig().Consensus
 
+	if consensusCfg.ServerModeEnabled {
```

Comment:
I agree with that, and in fact, I first did it like this. However, currently server mode is not managed through with the FSM. Because I faced a limitation with the FSM library when trying to transition from one state to two new states. So, if server mode is enabled, this must happen with the FSM:

After `StateMachineEvent_Start` :
- FSM must transition to `P2P_Bootstrapping` state (which continues as `P2P_Bootstrapped`).
- FSM must also transition to EnableServerMode state, which must be running in parallel to the all states. Therefore this state transition must happen mutually exclusive to the `P2P_Bootstrapping` state transition. However, there is an issue with transitioning to more than one states with the current lib.

I will be investigating this issue, and if I find a easy solution I will enable State_Machine events.